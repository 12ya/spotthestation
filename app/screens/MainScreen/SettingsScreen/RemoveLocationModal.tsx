import React from "react"
import { ViewStyle, View, PressableProps, TextStyle } from "react-native"
import { Icon, Text, Button } from "../../../components"
import { colors, typography } from "../../../theme"
import { LocationType } from "../../OnboardingScreen/SignupLocation"

export interface RemoveLocationModalProps {
  onClose?: PressableProps["onPress"]
  onRemove?: PressableProps["onPress"]
  location: LocationType
}

export function RemoveLocationModal({ onClose, onRemove, location }: RemoveLocationModalProps) {
  return (
    <View style={$modalBodyContainer}>
      <Icon icon="x" 
        accessible
        accessibilityLabel="x button"
        accessibilityHint="close modal"
        accessibilityRole="button"
        color={colors.palette.neutral450} 
        onPress={onClose} 
        containerStyle={$close} 
        size={36}
      />
      <View style={$generalIcon}>
        <Icon icon="removeLocation" />
      </View>
      <View style={$contentContainer}>
        <Text tx="settings.locationSettingsData.removeLocation.question" style={$title} />
        <Text text={location?.title} style={$locationTitle} />
        <Text text={location?.subtitle} style={$locationAddress} />
        
        <View style={$buttonsContainer}>
          <Button
            tx="settings.locationSettingsData.removeLocation.cancelButton"
            pressedStyle={$button}
            textStyle={$buttonText}
            style={$button}
            onPressIn={onClose}
          />
          <Button
            tx="settings.locationSettingsData.removeLocation.removeButton"
            pressedStyle={[$button, { backgroundColor: colors.palette.nasaRed }]}
            textStyle={$buttonText}
            style={[$button, { backgroundColor: colors.palette.nasaRed }]}
            onPressIn={onRemove}
          />
        </View>
      </View>
    </View>
  )
}

const $modalBodyContainer: ViewStyle = {
  backgroundColor: colors.palette.neutral350,
  borderTopLeftRadius: 18,
  borderTopRightRadius: 18,
}

const $buttonsContainer: ViewStyle = {
  flexDirection: 'row',
  alignItems: "center",
  justifyContent: "space-between",
}

const $contentContainer: ViewStyle = {
  width: "100%",
  paddingHorizontal: 36,
  paddingBottom: 52
}

const $close: ViewStyle = {
  position: "absolute",
  top: 0,
  right: 0,
  padding: 18,
  zIndex: 5
}

const $generalIcon: ViewStyle = {
  marginVertical: 36,
  width: "100%",
  alignItems: "center"
}

const $locationTitle: TextStyle = {
  textAlign: "center",
  color: colors.palette.neutral250,
  fontSize: 18,
  fontFamily: typography.primary.medium,
  lineHeight: 22,
  paddingBottom: 5
}

const $title: TextStyle = {
  textAlign: "center",
  color: colors.palette.neutral450,
  fontSize: 18,
  fontFamily: typography.primary.normal,
  lineHeight: 22,
  paddingBottom: 24
}

const $locationAddress: TextStyle = {
  textAlign: "center",
  color: colors.palette.neutral450,
  fontSize: 16,
  fontFamily: typography.primary.normal,
  lineHeight: 19,
  paddingBottom: 36
}

const $button: ViewStyle = {
  width: '40%',
  height: 64,
  backgroundColor: colors.palette.neutral550,
  borderRadius: 28,
  borderWidth: 0,
  marginTop: 24,
  marginBottom: 24
}

const $buttonText: TextStyle = {
  color: colors.palette.neutral100,
  fontSize: 18,
  fontFamily: typography.primary.medium,
  lineHeight: 22
}
