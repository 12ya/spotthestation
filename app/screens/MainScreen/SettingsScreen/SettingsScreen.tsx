import { useNavigation } from "@react-navigation/native"
import { observer } from "mobx-react-lite"
// import uniqBy from "lodash/uniqBy"
import React from "react"
import { View, ViewStyle, TextStyle } from "react-native"
import { useSafeAreaInsets } from "react-native-safe-area-context"
import { Screen, Text /* Icon */ } from "../../../components"
import { colors, typography, spacing } from "../../../theme"
import { SettingsItem } from "../components/SettingsItem"
import { ListItem } from "../components/ListItem"
import { useStores } from "../../../models"
import { translate /* setLocale */ } from "../../../i18n"
// import { Dropdown } from "react-native-element-dropdown"
import { StyleFn, useStyles } from "../../../utils/useStyles"
// import i18n from "i18n-js"
// import * as storage from "../../../utils/storage"

// const languages = uniqBy(
//   Object.keys(i18n.translations).map((key) => ({
//     label: (i18n.translations[key] as any).name as string,
//     value: key,
//   })),
//   "label",
// )

export const SettingsScreen = observer(function SettingsScreen() {
  const {
    $headerStyleOverride,
    $container,
    $itemsConteiner,
    $headerContainer,
    $header,
    // $dropdown,
    // $dropdownContainer,
    // $dropdownPlaceholder,
    // $dropdownSelected,
    // $dropdownText,
    // $dropdownRightAccessory,
  } = useStyles(styles)
  const navigation = useNavigation()
  const topInset = useSafeAreaInsets().top
  const { isLocalCalculations, setLocalCalculations /* setNotifications */ } = useStores()

  const handleNavigate = (screen) =>
    navigation.navigate("SettingsScreens" as never, { screen } as never)

  // const onChangeLanguage = async ({ value }) => {
  //   await storage.save("locale", value)
  //   setLocale(value as string)
  //   setNotifications()
  //   navigation.reset({
  //     index: 0,
  //     routes: [{ name: "Settings" as never }],
  //   })
  // }

  const headerStyle = { ...$headerStyleOverride }
  headerStyle.top = Number(headerStyle.top) + topInset

  return (
    <Screen
      preset="fixed"
      contentContainerStyle={[$container]}
      style={{ backgroundColor: colors.palette.neutral900 }}
      statusBarStyle="light"
    >
      <View style={[$headerContainer, headerStyle]}>
        <Text
          accessible
          accessibilityLabel="header"
          accessibilityHint="header"
          accessibilityRole="text"
          tx="settings.header"
          style={$header}
        />
      </View>
      <View style={$itemsConteiner}>
        <SettingsItem
          icon="mapPinOutlined"
          title="settings.locationSettings"
          onPress={() => handleNavigate("LocationSettings")}
        />
        <SettingsItem
          icon="bellRing"
          title="settings.notificationSettings"
          onPress={() => handleNavigate("NotificationSettings")}
        />
        <SettingsItem
          icon="shield"
          title="settings.termsAndConditions"
          onPress={() => handleNavigate("TermsAndConditions")}
        />
        <SettingsItem
          icon="mail"
          title="settings.contactUs"
          onPress={() => handleNavigate("ContactUs")}
        />
        {/* <SettingsItem */}
        {/*   icon="globe" */}
        {/*   title="settings.language" */}
        {/*   rightControl={ */}
        {/*     <Dropdown */}
        {/*       accessibilityLabel="language select" */}
        {/*       style={$dropdown} */}
        {/*       placeholderStyle={[$dropdownText, $dropdownPlaceholder]} */}
        {/*       selectedTextStyle={[$dropdownText, $dropdownSelected]} */}
        {/*       data={languages} */}
        {/*       itemContainerStyle={{ */}
        {/*         backgroundColor: colors.palette.neutral350, */}
        {/*       }} */}
        {/*       containerStyle={$dropdownContainer} */}
        {/*       itemTextStyle={$dropdownText} */}
        {/*       activeColor={colors.palette.neutral450} */}
        {/*       value={i18n.locale} */}
        {/*       labelField="label" */}
        {/*       valueField="value" */}
        {/*       onChange={onChangeLanguage} */}
        {/*       renderRightIcon={() => ( */}
        {/*         <Icon */}
        {/*           icon="chevronDown" */}
        {/*           size={28} */}
        {/*           color={colors.palette.neutral450} */}
        {/*           containerStyle={$dropdownRightAccessory} */}
        {/*         /> */}
        {/*       )} */}
        {/*     /> */}
        {/*   } */}
        {/* /> */}
        {/* eslint-disable-next-line react-native/no-inline-styles */}
        <View style={{ flex: 1 }} />
        <ListItem
          icon="iss"
          title={translate("settings.localCalculations")}
          subtitle=""
          selected={isLocalCalculations}
          onToggle={() => setLocalCalculations(!isLocalCalculations)}
          withSwitch
          borderless
        />
      </View>
    </Screen>
  )
})

const styles: StyleFn = ({ scale, fontSizes, lineHeights }) => {
  const $headerStyleOverride: TextStyle = {
    top: scale(24),
  }

  const $container: ViewStyle = {
    flex: 1,
    backgroundColor: colors.backgroundDark,
    justifyContent: "space-between",
  }

  const $itemsConteiner: ViewStyle = {
    flex: 1,
    marginTop: scale(150),
    paddingHorizontal: scale(36),
  }

  const $headerContainer: ViewStyle = {
    position: "absolute",
    left: scale(18),
    zIndex: 9,
  }

  const $header: TextStyle = {
    fontFamily: typography.primary.normal,
    fontSize: fontSizes[36],
    lineHeight: lineHeights[44],
    color: colors.palette.neutral250,
  }

  const $dropdown: ViewStyle = {
    flex: 1,
    height: scale(30),
    overflow: "hidden",
  }

  const $dropdownContainer: ViewStyle = {
    backgroundColor: colors.palette.neutral350,
    borderRadius: scale(10),
    marginTop: -scale(40),
    borderWidth: 0,
  }

  const $dropdownPlaceholder: TextStyle = {
    color: colors.palette.neutral450,
  }

  const $dropdownSelected: TextStyle = {
    color: colors.palette.neutral250,
  }

  const $dropdownText: TextStyle = {
    flex: 1,
    textAlign: "right",
    fontFamily: typography.primary.normal,
    fontSize: fontSizes[18],
    paddingVertical: 0,
    paddingHorizontal: 0,
    marginHorizontal: scale(spacing.small),
    textAlignVertical: "center",
    color: colors.palette.neutral250,
  }

  const $dropdownRightAccessory: ViewStyle = {
    height: scale(56),
    justifyContent: "center",
    alignItems: "center",
  }

  return {
    $headerStyleOverride,
    $container,
    $itemsConteiner,
    $headerContainer,
    $header,
    $dropdown,
    $dropdownContainer,
    $dropdownPlaceholder,
    $dropdownSelected,
    $dropdownText,
    $dropdownRightAccessory,
  }
}
