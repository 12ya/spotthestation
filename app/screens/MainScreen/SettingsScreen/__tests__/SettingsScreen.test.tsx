/* eslint-disable @typescript-eslint/no-unsafe-return */
/* eslint-disable react/display-name */
/* eslint-disable @typescript-eslint/no-unsafe-call */
/* eslint-disable @typescript-eslint/no-unsafe-member-access */
import React from "react"
import { useNavigation } from "@react-navigation/native"
import { SettingsScreen } from "../SettingsScreen"
import { fireEvent, render } from "@testing-library/react-native"

jest.mock("@react-navigation/native", () => ({
  useNavigation: jest.fn(),
  createNavigationContainerRef: jest.fn(),
}))

jest.mock("mobx-react-lite", () => ({
  observer: (component) => component,
}))

jest.mock("react-native-safe-area-context", () => ({
  useSafeAreaInsets: () => ({ top: 20 }),
}))

describe("SettingsScreen", () => {
  const mockNavigation = {
    navigate: jest.fn(),
  }

  beforeEach(() => {
    ;(useNavigation as any).mockReturnValue(mockNavigation)
  })

  afterEach(() => {
    jest.clearAllMocks()
  })

  test("renders correctly", () => {
    const tree = render(<SettingsScreen />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test("renders header correctly", () => {
    const { getByText } = render(<SettingsScreen />)
    const header = getByText("settings.header undefined")
    expect(header).toBeTruthy()
  })

  test("navigates to LocationSettings screen when location settings item is pressed", () => {
    const { getByText } = render(<SettingsScreen />)
    const locationSettingsItem = getByText("settings.locationSettings undefined")
    ;(fireEvent as any).press(locationSettingsItem)
    expect(mockNavigation.navigate).toHaveBeenCalledWith("SettingsScreens", {
      screen: "LocationSettings",
    })
  })
})
