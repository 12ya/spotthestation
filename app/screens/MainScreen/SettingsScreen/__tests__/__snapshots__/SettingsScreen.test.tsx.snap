// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsScreen renders correctly 1`] = `
<div>
  <View
    style={
      Array [
        Object {
          "left": 26,
          "position": "absolute",
          "zIndex": 9,
        },
        Object {
          "top": 55,
        },
      ]
    }
  >
    <Text
      accessibilityHint="header"
      accessibilityLabel="header"
      accessibilityRole="text"
      accessible={true}
      allowFontScaling={false}
      style={
        Array [
          Object {},
          Array [
            Object {
              "fontSize": 23,
              "lineHeight": 35,
            },
            Object {
              "fontFamily": "Helvetica Neue",
            },
            Object {
              "color": "#191015",
            },
          ],
          undefined,
          undefined,
          Object {
            "color": "#DEDDDE",
            "fontFamily": "Helvetica Neue",
            "fontSize": 52,
            "lineHeight": 63,
          },
        ]
      }
    >
      settings.header undefined
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "marginTop": 215,
        "paddingHorizontal": 52,
      }
    }
  >
    <View
      accessibilityHint="pressable settings item"
      accessibilityLabel="pressable settings item"
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "flex-start",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingTop": 23,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-start",
              "borderBottomWidth": 2,
              "borderColor": "#353739",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginLeft": 15,
              "paddingBottom": 23,
            },
            false,
          ]
        }
      >
        <View>
          <RCTImageView
            internal_analyticTag={null}
            resizeMode="contain"
            source={
              Array [
                Object {
                  "height": undefined,
                  "uri": undefined,
                  "width": undefined,
                },
              ]
            }
            style={
              Object {
                "height": 35,
                "overflow": "hidden",
                "resizeMode": "contain",
                "tintColor": "#ADADAE",
                "width": 35,
              }
            }
            tintColor="#ADADAE"
          />
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "column",
              "paddingLeft": 26,
            }
          }
        >
          <Text
            accessibilityHint="settings.locationSettings"
            accessibilityLabel="title"
            accessibilityRole="text"
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {},
                Array [
                  Object {
                    "fontSize": 23,
                    "lineHeight": 35,
                  },
                  Object {
                    "fontFamily": "Helvetica Neue",
                  },
                  Object {
                    "color": "#191015",
                  },
                ],
                undefined,
                undefined,
                Object {
                  "color": "#FFFFFF",
                  "fontFamily": "Helvetica Neue",
                  "fontSize": 35,
                  "lineHeight": 42,
                  "width": "95%",
                },
              ]
            }
          >
            settings.locationSettings undefined
          </Text>
        </View>
        <View>
          <RCTImageView
            internal_analyticTag={null}
            resizeMode="contain"
            source={
              Array [
                Object {
                  "height": undefined,
                  "uri": undefined,
                  "width": undefined,
                },
              ]
            }
            style={
              Object {
                "height": 35,
                "overflow": "hidden",
                "resizeMode": "contain",
                "tintColor": "#353739",
                "width": 35,
              }
            }
            tintColor="#353739"
          />
        </View>
      </View>
    </View>
    <View
      accessibilityHint="pressable settings item"
      accessibilityLabel="pressable settings item"
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "flex-start",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingTop": 23,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-start",
              "borderBottomWidth": 2,
              "borderColor": "#353739",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginLeft": 15,
              "paddingBottom": 23,
            },
            false,
          ]
        }
      >
        <View>
          <RCTImageView
            internal_analyticTag={null}
            resizeMode="contain"
            source={
              Array [
                Object {
                  "height": undefined,
                  "uri": undefined,
                  "width": undefined,
                },
              ]
            }
            style={
              Object {
                "height": 35,
                "overflow": "hidden",
                "resizeMode": "contain",
                "tintColor": "#ADADAE",
                "width": 35,
              }
            }
            tintColor="#ADADAE"
          />
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "column",
              "paddingLeft": 26,
            }
          }
        >
          <Text
            accessibilityHint="settings.notificationSettings"
            accessibilityLabel="title"
            accessibilityRole="text"
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {},
                Array [
                  Object {
                    "fontSize": 23,
                    "lineHeight": 35,
                  },
                  Object {
                    "fontFamily": "Helvetica Neue",
                  },
                  Object {
                    "color": "#191015",
                  },
                ],
                undefined,
                undefined,
                Object {
                  "color": "#FFFFFF",
                  "fontFamily": "Helvetica Neue",
                  "fontSize": 35,
                  "lineHeight": 42,
                  "width": "95%",
                },
              ]
            }
          >
            settings.notificationSettings undefined
          </Text>
        </View>
        <View>
          <RCTImageView
            internal_analyticTag={null}
            resizeMode="contain"
            source={
              Array [
                Object {
                  "height": undefined,
                  "uri": undefined,
                  "width": undefined,
                },
              ]
            }
            style={
              Object {
                "height": 35,
                "overflow": "hidden",
                "resizeMode": "contain",
                "tintColor": "#353739",
                "width": 35,
              }
            }
            tintColor="#353739"
          />
        </View>
      </View>
    </View>
    <View
      accessibilityHint="pressable settings item"
      accessibilityLabel="pressable settings item"
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "flex-start",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingTop": 23,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-start",
              "borderBottomWidth": 2,
              "borderColor": "#353739",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginLeft": 15,
              "paddingBottom": 23,
            },
            false,
          ]
        }
      >
        <View>
          <RCTImageView
            internal_analyticTag={null}
            resizeMode="contain"
            source={
              Array [
                Object {
                  "height": undefined,
                  "uri": undefined,
                  "width": undefined,
                },
              ]
            }
            style={
              Object {
                "height": 35,
                "overflow": "hidden",
                "resizeMode": "contain",
                "tintColor": "#ADADAE",
                "width": 35,
              }
            }
            tintColor="#ADADAE"
          />
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "column",
              "paddingLeft": 26,
            }
          }
        >
          <Text
            accessibilityHint="settings.termsAndConditions"
            accessibilityLabel="title"
            accessibilityRole="text"
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {},
                Array [
                  Object {
                    "fontSize": 23,
                    "lineHeight": 35,
                  },
                  Object {
                    "fontFamily": "Helvetica Neue",
                  },
                  Object {
                    "color": "#191015",
                  },
                ],
                undefined,
                undefined,
                Object {
                  "color": "#FFFFFF",
                  "fontFamily": "Helvetica Neue",
                  "fontSize": 35,
                  "lineHeight": 42,
                  "width": "95%",
                },
              ]
            }
          >
            settings.termsAndConditions undefined
          </Text>
        </View>
        <View>
          <RCTImageView
            internal_analyticTag={null}
            resizeMode="contain"
            source={
              Array [
                Object {
                  "height": undefined,
                  "uri": undefined,
                  "width": undefined,
                },
              ]
            }
            style={
              Object {
                "height": 35,
                "overflow": "hidden",
                "resizeMode": "contain",
                "tintColor": "#353739",
                "width": 35,
              }
            }
            tintColor="#353739"
          />
        </View>
      </View>
    </View>
    <View
      accessibilityHint="pressable settings item"
      accessibilityLabel="pressable settings item"
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "flex-start",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingTop": 23,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-start",
              "borderBottomWidth": 2,
              "borderColor": "#353739",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginLeft": 15,
              "paddingBottom": 23,
            },
            false,
          ]
        }
      >
        <View>
          <RCTImageView
            internal_analyticTag={null}
            resizeMode="contain"
            source={
              Array [
                Object {
                  "height": undefined,
                  "uri": undefined,
                  "width": undefined,
                },
              ]
            }
            style={
              Object {
                "height": 35,
                "overflow": "hidden",
                "resizeMode": "contain",
                "tintColor": "#ADADAE",
                "width": 35,
              }
            }
            tintColor="#ADADAE"
          />
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "column",
              "paddingLeft": 26,
            }
          }
        >
          <Text
            accessibilityHint="settings.contactUs"
            accessibilityLabel="title"
            accessibilityRole="text"
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {},
                Array [
                  Object {
                    "fontSize": 23,
                    "lineHeight": 35,
                  },
                  Object {
                    "fontFamily": "Helvetica Neue",
                  },
                  Object {
                    "color": "#191015",
                  },
                ],
                undefined,
                undefined,
                Object {
                  "color": "#FFFFFF",
                  "fontFamily": "Helvetica Neue",
                  "fontSize": 35,
                  "lineHeight": 42,
                  "width": "95%",
                },
              ]
            }
          >
            settings.contactUs undefined
          </Text>
        </View>
        <View>
          <RCTImageView
            internal_analyticTag={null}
            resizeMode="contain"
            source={
              Array [
                Object {
                  "height": undefined,
                  "uri": undefined,
                  "width": undefined,
                },
              ]
            }
            style={
              Object {
                "height": 35,
                "overflow": "hidden",
                "resizeMode": "contain",
                "tintColor": "#353739",
                "width": 35,
              }
            }
            tintColor="#353739"
          />
        </View>
      </View>
    </View>
    <View
      accessibilityHint="pressable settings item"
      accessibilityLabel="pressable settings item"
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "flex-start",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingTop": 23,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-start",
              "borderBottomWidth": 2,
              "borderColor": "#353739",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginLeft": 15,
              "paddingBottom": 23,
            },
            false,
          ]
        }
      >
        <View>
          <RCTImageView
            internal_analyticTag={null}
            resizeMode="contain"
            source={
              Array [
                Object {
                  "height": undefined,
                  "uri": undefined,
                  "width": undefined,
                },
              ]
            }
            style={
              Object {
                "height": 35,
                "overflow": "hidden",
                "resizeMode": "contain",
                "tintColor": "#ADADAE",
                "width": 35,
              }
            }
            tintColor="#ADADAE"
          />
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "column",
              "paddingLeft": 26,
            }
          }
        >
          <Text
            accessibilityHint="settings.language"
            accessibilityLabel="title"
            accessibilityRole="text"
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {},
                Array [
                  Object {
                    "fontSize": 23,
                    "lineHeight": 35,
                  },
                  Object {
                    "fontFamily": "Helvetica Neue",
                  },
                  Object {
                    "color": "#191015",
                  },
                ],
                undefined,
                undefined,
                Object {
                  "color": "#FFFFFF",
                  "fontFamily": "Helvetica Neue",
                  "fontSize": 35,
                  "lineHeight": 42,
                  "width": "95%",
                },
              ]
            }
          >
            settings.language undefined
          </Text>
        </View>
        <View
          onLayout={[Function]}
          style={
            Object {
              "flex": 1,
              "height": 43,
              "justifyContent": "center",
              "overflow": "hidden",
            }
          }
        >
          <View
            accessibilityLabel="language select"
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 35,
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "flex": 1,
                    "fontSize": 16,
                  },
                  Array [
                    Object {
                      "color": "#DEDDDE",
                      "flex": 1,
                      "fontFamily": "Helvetica Neue",
                      "fontSize": 26,
                      "marginHorizontal": 17,
                      "paddingHorizontal": 0,
                      "paddingVertical": 0,
                      "textAlign": "right",
                      "textAlignVertical": "center",
                    },
                    Object {
                      "color": "#ADADAE",
                    },
                  ],
                  Object {},
                ]
              }
            >
              Select item
            </Text>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "height": 80,
                  "justifyContent": "center",
                }
              }
            >
              <RCTImageView
                internal_analyticTag={null}
                resizeMode="contain"
                source={
                  Array [
                    Object {
                      "height": undefined,
                      "uri": undefined,
                      "width": undefined,
                    },
                  ]
                }
                style={
                  Object {
                    "height": 40,
                    "overflow": "hidden",
                    "resizeMode": "contain",
                    "tintColor": "#ADADAE",
                    "width": 40,
                  }
                }
                tintColor="#ADADAE"
              />
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    />
    <View
      accessibilityHint="pressable list item"
      accessibilityLabel="pressable list item"
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "flex-start",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingTop": 23,
          "width": "100%",
        }
      }
    >
      <View>
        <View>
          <RCTImageView
            internal_analyticTag={null}
            resizeMode="contain"
            source={
              Array [
                Object {
                  "height": undefined,
                  "uri": undefined,
                  "width": undefined,
                },
              ]
            }
            style={
              Object {
                "height": 35,
                "overflow": "hidden",
                "resizeMode": "contain",
                "tintColor": "#ADADAE",
                "width": 35,
              }
            }
            tintColor="#ADADAE"
          />
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-start",
              "borderBottomWidth": 2,
              "borderColor": "#353739",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginLeft": 15,
              "paddingBottom": 23,
            },
            Object {
              "borderBottomWidth": 0,
            },
          ]
        }
      >
        <View
          accessibilityHint="list item body"
          accessibilityLabel="list item body"
          accessibilityRole="text"
          accessible={true}
          style={
            Object {
              "flex": 1,
              "flexDirection": "column",
            }
          }
        >
          <Text
            allowFontScaling={false}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {},
                Array [
                  Object {
                    "fontSize": 23,
                    "lineHeight": 35,
                  },
                  Object {
                    "fontFamily": "Helvetica Neue",
                  },
                  Object {
                    "color": "#191015",
                  },
                ],
                undefined,
                undefined,
                Object {
                  "color": "#FFFFFF",
                  "fontFamily": "Helvetica Neue",
                  "fontSize": 26,
                  "lineHeight": 32,
                  "width": "95%",
                },
              ]
            }
          >
            settings.localCalculations undefined
          </Text>
          <Text
            allowFontScaling={false}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {},
                Array [
                  Object {
                    "fontSize": 23,
                    "lineHeight": 35,
                  },
                  Object {
                    "fontFamily": "Helvetica Neue",
                  },
                  Object {
                    "color": "#191015",
                  },
                ],
                undefined,
                undefined,
                Object {
                  "color": "#FFFFFF",
                  "fontFamily": "Helvetica Neue",
                  "fontSize": 23,
                  "lineHeight": 27,
                  "paddingTop": 7,
                  "width": "95%",
                },
              ]
            }
          />
        </View>
        <View
          accessibilityHint="toggle location alerts"
          accessibilityLabel="switch button"
          accessibilityRole="switch"
          accessibilityState={
            Object {
              "checked": true,
              "disabled": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderRadius": 23,
                    "borderWidth": 0,
                    "flexDirection": "row",
                    "flexGrow": 0,
                    "flexShrink": 0,
                    "height": 40,
                    "justifyContent": "space-between",
                    "overflow": "hidden",
                    "width": 66,
                  },
                  Object {
                    "backgroundColor": "#353739",
                  },
                  undefined,
                ]
              }
            >
              <View
                animatedStyle={
                  Object {
                    "value": Object {
                      "opacity": 1,
                    },
                  }
                }
                collapsable={false}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#0B3D91",
                    "borderColor": "rgba(0, 0, 0, 0)",
                    "height": "100%",
                    "opacity": 1,
                    "overflow": "hidden",
                    "paddingEnd": 6,
                    "paddingStart": 6,
                    "position": "absolute",
                    "width": "100%",
                  }
                }
              />
              <View
                animatedStyle={
                  Object {
                    "value": Object {
                      "marginStart": -35,
                      "start": "100%",
                    },
                  }
                }
                collapsable={false}
                style={
                  Object {
                    "backgroundColor": "#FFFFFF",
                    "borderRadius": 17,
                    "height": 29,
                    "marginStart": -35,
                    "position": "absolute",
                    "start": "100%",
                    "width": 29,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</div>
`;
